students/csdotson/lesson04/trigrams.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+# Lesson 4 - Trigrams
+import random
+
+def run():
+    input_file = input("Please enter the name of a file (with extension):\n")
+    read_file(input_file)
```

Comment:
I would suggest a better flow for your main entry point.

right now looking here last logic is to read a file, doesn't really tell me anything else is going on.. and then each following function calls next function. it is hard to follow and you don't get a full view at what the code is doing.

what you could do:
```
text = read_file(input_file)
story = make_story(text) 
print("here is your story: ", story)
```


```
def make_story(text):
    text_hash = create_hash(text)
    ....
    return fix_punctuation_and_display(story)
```